core/sr-primitives/src/transaction_validity.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2017 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Transaction validity interface.
+
+use rstd::prelude::*;
+
+/// Priority for a transaction. Additive. Higher is better.
+pub type TransactionPriority = u64;
+
+/// Tag for a transaction. No two transactions with the same tag should be placed on-chain.
+pub type TransactionTag = Vec<u8>;
+
+/// Information on a transaction's validity and, if valid, on how it relates to other transactions.
+#[derive(Clone, PartialEq, Eq, Encode, Decode)]
+pub enum TransactionValidity {
+	Invalid,
+	Valid(TransactionPriority, Vec<TransactionTag>, Vec<TransactionTag>),
```

Comment:
I would actually use a named fields here, cause we might easily forget which one was `requires` and which one `provides`.